src/util.cc

```diff
@@ -289,18 +307,18 @@ void GetShellEscapedString(const string& input, string* result) {
   }
 
   const char kQuote = '\'';
-  const char kEscapeSequence[] = "'\\'";
+  const char kEscapeSequence[] = "'\\''";
+
+  result->reserve(result->size() + input.size() + 2 * sizeof(kQuote));
 
   result->push_back(kQuote);
 
   string::const_iterator span_begin = input.begin();
-  for (string::const_iterator it = input.begin(), end = input.end(); it != end;
-       ++it) {
-    if (*it == kQuote) {
-      result->append(span_begin, it);
-      result->append(kEscapeSequence);
-      span_begin = it;
-    }
+  string::const_iterator it = std::find(span_begin, input.end(), kQuote);
```

Comment:
it is not used outside the scope of the for loop.

Consider putting this line as the initialization statement of the forloop.

```cpp
for (string::const_iterator it = std::find(span_begin, input.end(), kQuote);
      it != input.end();
      it = std::find(span_begin, input.end(), kQuote)) {
    result->append(span_begin, it);
    result->append(kEscapeSequence);
    span_begin = ++it;
}
```